be/src/storage/olap_server.cpp

```diff
@@ -78,7 +78,7 @@ Status StorageEngine::start_bg_threads() {
     for (auto& tmp_store : _store_map) {
         data_dirs.push_back(tmp_store.second);
     }
-    int32_t data_dir_num = data_dirs.size();
+    auto data_dir_num = static_cast<int32_t>(data_dirs.size());
```

Comment:
use const to indicate constness if the variable does not need to change may improve readability.